enhancements/agent-installer/interactive-network-config.md

```diff
@@ -0,0 +1,238 @@
+  - "@andfasano"
+  - "@rwsu"
+  - "@bfournie"
+  - "@cgwalters"
+  - "@patrickdillon"
+approvers:
+  - "@celebdor"
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - None
+creation-date: 2023-01-06
+last-updated: 2023-01-06
+tracking-link:
+  - https://issues.redhat.com/browse/AGENT-385
+see-also:
+  - "/enhancements/agent-installer/automated-workflow-for-agent-based-installer.md"
+replaces: N/A
+superseded-by: N/A
+---
+
+# Interactive Network Configuration
+
+## Summary
+
+Enable the user, after booting the agent ISO, to interactively add or correct
+any neccessary network configuration. Since we cannot rely on network
+connectivity for this step, configuration will be via a text user interface
+(TUI) displayed on the console.
+
+## Motivation
+
+Providing network configuration up-front, in the absence of the actual hosts to
+be used, is onerous for users. Names of NIC interfaces are difficult to predict
+in advance, and often MAC addresses are required. Even the simplified NMState
+format is difficult to write by hand.
+
+Correcting any error in the configuration passed into the installer currently
+requires returning to the beginning of the process, generating a new ISO and
+booting the hosts anew with the updated image. There is no guarantee of getting
+it right, so there could be multiple iterations of this process.
+
+### User Stories
+
+* As a cluster administrator, I want to install a cluster into a network
+  environment requiring custom network configuration (e.g. static IPs, NIC
+  bonds, VLANs) without needing any information from the physical servers in
+  advance.
+* As a cluster administrator, I want to fix a cluster installation that has
+  stalled due to incorrect network configuration without having to regenerate
+  the installation image.
+
+### Goals
+
+* Allow the user to configure static IPs, NIC bonds, and VLANs.
+* Make available all necessary information about the hardware and network
+  environemnt to do this configuration.
+* Provide an indication of when sufficient connectivity is or is not available.
+* Enable the user to debug connectivity problems by narrowing down the cause
+  (DNS, reachability, &c.).
+* Work for both the existing automation workflow and a future fully-interactive
+  workflow.
+
+### Non-Goals
+
+* Interactive (re)configuration of any other aspect of the installation.
+* Add interactive network configuration to CoreOS in general.
+* Allow interactive network configuration during IPI baremetal installs.
+
+## Proposal
+
+### Workflow Description
+
+The user will generate the agent ISO using `openshift-install` as usual. The
+`networkConfig` section in the `agent-config` (or the NMStateConfig ZTP
+manifests) may _or may not_ contain network configuration.
+
+The user will then boot the ISO. Any network configuration for each current
```

Comment:
If the agent-config contains a hostname for the host, it should be applied before bringing up the TUI. Is this correct?

Reply:
Yes, I think it should. Users would probably be confused if they configured hostnames and then those weren't applied when they started manually editing the config.